script/send_new_webmentions

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env ruby
+
+ENV["RAILS_ENV"] ||= "production"
+
+APP_PATH = File.expand_path('../../config/application', __FILE__)
+require File.expand_path('../../config/boot', __FILE__)
+require APP_PATH
+Rails.application.require_environment!
+
+LAST_STORY_KEY = "webmentions:last_story_id".freeze
+
+def endpoint_from_body(html)
+  doc = Nokogiri::HTML(html)
+
+  if !doc.css('[rel~="webmention"]').css('[href]').empty?
+    doc.css('[rel~="webmention"]').css('[href]').attribute('href').value
+  elsif !doc.css('[rel="http://webmention.org/"]').css('[href]').empty?
+    doc.css('[rel="http://webmention.org/"]').css('[href]').attribute('href').value
+  elsif !doc.css('[rel="http://webmention.org"]').css('[href]').empty?
+    doc.css('[rel="http://webmention.org"]').css('[href]').attribute('href').value
+  end
+end
+
+def endpoint_from_headers(header)
+  return unless header
+
+  if (matches = header.match(/<([^>]+)>; rel="[^"]*\s?webmention\s?[^"]*"/))
+    return matches[1]
+  elsif (matches = header.match(/<([^>]+)>; rel=webmention/))
+    return matches[1]
+  elsif (matches = header.match(/rel="[^"]*\s?webmention\s?[^"]*"; <([^>]+)>/))
+    return matches[1]
+  elsif (matches = header.match(/rel=webmention; <([^>]+)>/))
+    return matches[1]
+  elsif (matches = header.match(/<([^>]+)>; rel="http:\/\/webmention\.org\/?"/))
+    return matches[1]
+  elsif (matches = header.match(/rel="http:\/\/webmention\.org\/?"; <([^>]+)>/))
+    return matches[1]
+  end
+end
+
+# Some pages could return a relative link as their webmention endpoint.
+# We need to translate this relative likn to an absolute one.
+def uri_to_absolute(uri, req_uri)
+  abs_uri = URI.parse(uri)
+  if abs_uri.host
+    # Already absolute.
+    return uri
+  else
+    abs_uri.host = req_uri.host
+    abs_uri.scheme = req_uri.scheme
+    abs_uri.port = req_uri.port
+    return abs_uri
+  end
+end
+
+def send_webmention(source, target, endpoint)
+  sp = Sponge.new
+  sp.timeout = 5
+  sp.fetch(endpoint.to_s, :post, {
+    "source" => URI.encode_www_form_component(source),
+    "target" => URI.encode_www_form_component(target),
+  }, nil, {}, 3)
+end
+
+last_story_id = (Keystore.value_for(LAST_STORY_KEY) || Story.last.id).to_i
+Story.where("id > ? AND is_expired = ?", last_story_id, false).order(:id).each do |s|
+  if s.url.present?
```

Comment:
`next unless s.url.present?` to save an indent and save me looking for an `else` :) Same with the other two `if`s.